### PR TITLE
Slurm on GCP slurm-controller-hybrid module wrapper

### PR DESCRIPTION
**Pull request by heyealex**
This PR adds a wrapper to the slurm-gcp slurm-controller-hybrid module. A test_config using this module has also been added.

### Submission Checklist

* [x] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [x] Are all tests passing? (`make tests`)
* [x] Have you written unit tests to cover this change?
* [x] Is unit test coverage still above 80%?
* [x] Have you updated all applicable documentation?
* [x] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

File: community/modules/scheduler/schedmd-slurm-gcp-v5-hybrid/README.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+[slurm-controller-hybrid]: https://github.com/SchedMD/slurm-gcp/tree/v5.0.3/terraform/slurm_cluster/modules/slurm_controller_hybrid
+
+> **_NOTE:_** The hybrid module requires the following dependencies to be
+> installed on the system deploying the module:
+>
+> * [terraform]
+> * [addict]
+> * [google-cloud-pubsub]
+> * A full list of recommended python packages is available in a
+>   [requirements.txt] file in the [slurm-gcp] repo.
```
Comment (nick-stroud) [inline community/modules/scheduler/schedmd-slurm-gcp-v5-hybrid/README.md:38]: When testing this in a clean virtual environment I needed to install the following:
```
pip install addict httplib2 google-api-python-client pyyaml
```
I have confirmed that httplib2 and pyyaml are not dependencies of `google-cloud-pubsub`, I would recommend updating the list to what I observed.

File: community/modules/scheduler/schedmd-slurm-gcp-v5-hybrid/README.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+The expectation is that these directories exist on the controller and that all
+files required by slurmd to be in sync with the controller are in those
+directories.
+
+If this does not match your slurm cluster, these directories can be overwritten
+with a custom NFS mount using [pre-existing-network-storage] or by setting the
+`network_storage` variable directly in the hybrid module. Any value in
+`network_storage`, added directly or with `use`, will override the default
+directories above.
```
Comment (nick-stroud) [inline community/modules/scheduler/schedmd-slurm-gcp-v5-hybrid/README.md:105]: nit: this last line maybe should be in bold.

File: community/modules/scheduler/schedmd-slurm-gcp-v5-hybrid/variables.tf [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+    server_ip     = string
+    remote_mount  = string
+    local_mount   = string
+    fs_type       = string
+    mount_options = string
+  }))
+  default = []
+}
+
+variable "login_network_storage" {
```
Comment (nick-stroud) [inline community/modules/scheduler/schedmd-slurm-gcp-v5-hybrid/variables.tf:160]: from offline discussion: delete.

File: community/modules/scheduler/schedmd-slurm-gcp-v5-hybrid/README.md [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+as part of the [slurm-gcp] github repository. The hybrid module serves to create
+the configurations needed to extend an on-premise slurm cluster to one with one
+or more Google Cloud bursting partitions. These partitions will create the
+requested nodes in a GCP project on-demand and scale after a period of not being
+used, in the same way as the [schedmd-slurm-gcp-v5-controller] module
+auto-scales VMs.
+
+> **_NOTE:_** This module has only been tested in limited capacity with the HPC
+> Toolkit. On Premise configurations can vary significantly, this module should
+> be used as a starting point, not a complete solution.
```
Comment (nick-stroud) [inline community/modules/scheduler/schedmd-slurm-gcp-v5-hybrid/README.md:12]: I would explicitly add that this module is `experimental` and that it is expected that the module and documentation may change significantly in the future. 

Also add it to the list of modules, that may have gotten lost in the refactor.

**Review by nick-stroud — COMMENTED** [review]
Given the disclaimer that

File: community/modules/scheduler/schedmd-slurm-gcp-v5-hybrid/README.md (hunk 1 above)
Comment (heyealex) [reply community/modules/scheduler/schedmd-slurm-gcp-v5-hybrid/README.md:43]: Added

File: community/modules/scheduler/schedmd-slurm-gcp-v5-hybrid/README.md (hunk 2 above)
Comment (heyealex) [reply community/modules/scheduler/schedmd-slurm-gcp-v5-hybrid/README.md]: Done

File: community/modules/scheduler/schedmd-slurm-gcp-v5-hybrid/variables.tf (hunk 3 above)
Comment (heyealex) [reply community/modules/scheduler/schedmd-slurm-gcp-v5-hybrid/variables.tf]: Done

File: community/modules/scheduler/schedmd-slurm-gcp-v5-hybrid/README.md (hunk 4 above)
Comment (heyealex) [reply community/modules/scheduler/schedmd-slurm-gcp-v5-hybrid/README.md:14]: Done

File: community/modules/scheduler/schedmd-slurm-gcp-v5-hybrid/README.md (hunk 1 above)
Comment (nick-stroud) [reply community/modules/scheduler/schedmd-slurm-gcp-v5-hybrid/README.md:43]: nit: it is not clear that terraform is the terraform binary and the rest are python pip packages. We can clean up in our next pass.

**Review by nick-stroud — APPROVED** [review]

**Pull request merged**